unittests/CppInterOp/DynamicLibraryManagerTest.cpp

```diff
@@ -2,55 +2,216 @@
 
 #include "clang/Interpreter/CppInterOp.h"
 
+
+#include "llvm/ExecutionEngine/Orc/Core.h"
+#include "llvm/ExecutionEngine/Orc/DebugUtils.h"
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/Path.h"
+#include "llvm/Support/raw_ostream.h"
+
+// Helper functions
 
 // This function isn't referenced outside its translation unit, but it
 // can't use the "static" keyword because its address is used for
 // GetMainExecutable (since some platforms don't support taking the
 // address of main, and some platforms can't implement GetMainExecutable
 // without being given the address of a function in the main executable).
-std::string GetExecutablePath(const char* Argv0) {
+std::string GetExecutablePath(const char* Argv0, void* mainAddr = nullptr) {
   // This just needs to be some symbol in the binary; C++ doesn't
   // allow taking the address of ::main however.
-  void* MainAddr = (void*)intptr_t(GetExecutablePath);
+  void* MainAddr = mainAddr ? mainAddr : (void*)intptr_t(GetExecutablePath);
   return llvm::sys::fs::getMainExecutable(Argv0, MainAddr);
 }
 
+// Mangle symbol name
+static inline std::string MangleNameForDlsym(const std::string& name) {
+  std::string nameForDlsym = name;
+#if defined(R__MACOSX) || defined(R__WIN32)
+  if (nameForDlsym[0] != '_')
+    nameForDlsym.insert (0, 1, '_');
+#endif //R__MACOSX
+  return nameForDlsym;
+}
+
+#include "../../lib/Interpreter/CppInterOp.cpp"
```

Comment:
We probably should drop this.

Reply:
We use from it functions that are not visible through the .h file ie. should be available only for the tests.

Reply:
Okay, fair enough...